### PR TITLE
Restrictions settings

### PR DESCRIPTION
**Pull request by jpdimond**
Provides a settings panel in meals for community dietary restrictions with feature flag.

### CODE REVIEW

File: app/controllers/meals/restrictions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    def update
+      @community = Community.find(params[:id])
+      authorize(@community)
+      begin
+        result = @community.update!(community_params)
+        flash[:success] = "Updated successfully."
+        render(:edit)
+      rescue => e
+        prep_vars
+        flash[:error] = e
```
Comment (jpdimond) [inline app/controllers/meals/restrictions_controller.rb:20]: Would like to pair on this, the errors aren't showing up inline on the form

File: app/javascript/controllers/index.ts [hunk 2]
```diff
@@ -17,4 +17,4 @@ import PartialListController from "./partial_list_controller"
 application.register("partial-list", PartialListController)
 
 import PaymentController from "./payment_controller"
-application.register("payment", PaymentController)
+application.register("payment", PaymentController
```
Comment (smoyte) [inline app/javascript/controllers/index.ts:20]: What's going on here?

Comment (smoyte) [reply app/javascript/controllers/index.ts:20]: Can you also please delete empty restrictions_controller.ts 🙏

File: app/views/meals/restrictions/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<% content_for(:title, "Meals Settings: Dietary Restrictions") %>
+<p>
+  Please enter the dietary restriction by how it is described when a meal 
+  contains it, and when there is an abscence of it, as both forms are used. 
+  We have given common restrictions.
```
Comment (smoyte) [inline app/views/meals/restrictions/edit.html.erb:5]: Can you remove `We have given common restrictions.` please? I don't think it is necessary and it may be confusing over time who "we" refers to

File: spec/system/meals/restrictions_spec.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  scenario "happy path" do
+    visit(edit_meals_restrictions_path)
+
+    click_link("Add restriction")
+    fill_in("Contains", with: "gluten")
+    fill_in("Absence", with: "no gluten")
+    click_button("Save")
+
+    expect(page).to have_success_alert("Updated successfully")
+    expect(page).to have_field("Contains", with: "gluten")
```
Comment (smoyte) [inline spec/system/meals/restrictions_spec.rb:22]: Nice!!!

**Review by smoyte — CHANGES_REQUESTED** [review]
Awesome! Just a few small things. Let's pair to fix your error issue. I'd also like to see how it looks!

**Review by smoyte — CHANGES_REQUESTED** [review]
Almost there! Thanks for the updates!

Can you just get rid of the deactivated_at column and the Deactivatable include now? Or is it doing something useful? If you want to keep it then I think you'd need some unit tests on setting deactivated_at automatically. I think there may be some edge cases not covered, like when you re-activate something. But yeah not sure you need it.

**Comment by smoyte** [comment]
Can you also share a screenshot of the final UI please?

**Comment by jpdimond** [comment]
> Can you also share a screenshot of the final UI please?

image

File: db/migrate/20250218191831_create_meal_restrictions.rb [hunk 5]
```diff
@@ -5,6 +5,7 @@ def change
       t.string :absence, null: false, limit: 64
       t.references :cluster, foreign_key: true, index: true, null: false
       t.datetime :deactivated_at
```
[the remark this answers is not in the document]
Comment (jpdimond) [reply db/migrate/20250218191831_create_meal_restrictions.rb:7]: removed

File: app/models/meals/restriction.rb [hunk 6]
```diff
@@ -2,17 +2,23 @@ module Meals
...
+    before_validation :set_deactivated_at
+
+    def deactivated?
+      deactivated
     end
 
+    private 
+    def set_deactivated_at
+      self.deactivated_at = DateTime.now if deactivated?
+    end
```
[the remark this answers is not in the document]
Comment (jpdimond) [reply app/models/meals/restriction.rb:20]: removed

File: app/models/meals/restriction.rb [hunk 7]
```diff
@@ -2,17 +2,23 @@ module Meals
   class Restriction < ApplicationRecord
     include Deactivatable
```
[the remark this answers is not in the document]
Comment (jpdimond) [reply app/models/meals/restriction.rb:3]: removed

**Review by smoyte — CHANGES_REQUESTED** [review]
image

🙏

**Comment by smoyte** [comment]
I also just noticed you have a bunch of test failures?
